engine/table/src/main/java/io/deephaven/engine/table/impl/updateby/rollingwavg/BasePrimitiveRollingWAvgOperator.java

```diff
@@ -0,0 +1,198 @@
+package io.deephaven.engine.table.impl.updateby.rollingwavg;
+
+import io.deephaven.base.ringbuffer.AggregatingDoubleRingBuffer;
+import io.deephaven.base.verify.Assert;
+import io.deephaven.chunk.*;
+import io.deephaven.chunk.attributes.Values;
+import io.deephaven.engine.table.MatchPair;
+import io.deephaven.engine.table.impl.updateby.internal.BaseDoubleUpdateByOperator;
+import io.deephaven.engine.table.impl.util.RowRedirection;
+import io.deephaven.function.Basic;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+
+import static io.deephaven.util.QueryConstants.NULL_DOUBLE;
+
+abstract class BasePrimitiveRollingWAvgOperator extends BaseDoubleUpdateByOperator {
+    private static final int BUFFER_INITIAL_CAPACITY = 128;
+    final String weightColumnName;
+    final ChunkType weightChunkType;
+    final Class<?> weightColumnSourceType;
+
+    // region extra-fields
+    // endregion extra-fields
+
+    abstract class Context extends BaseDoubleUpdateByOperator.Context {
+        WritableDoubleChunk<? extends Values> influencerWeightValuesChunk;
+        AggregatingDoubleRingBuffer windowValues;
+        AggregatingDoubleRingBuffer windowWeightValues;
+
+        protected Context(final int affectedChunkSize, final int influencerChunkSize) {
+            super(affectedChunkSize);
+            windowValues = new AggregatingDoubleRingBuffer(BUFFER_INITIAL_CAPACITY, 0.0, (a, b) -> {
+                if (a == NULL_DOUBLE) {
+                    return b;
+                } else if (b == NULL_DOUBLE) {
+                    return a;
+                }
+                return a + b;
+            });
+            windowWeightValues = new AggregatingDoubleRingBuffer(BUFFER_INITIAL_CAPACITY, 0.0, (a, b) -> {
+                if (a == NULL_DOUBLE) {
+                    return b;
+                } else if (b == NULL_DOUBLE) {
+                    return a;
+                }
+                return a + b;
+            });
+            influencerWeightValuesChunk = WritableDoubleChunk.makeWritableChunk(influencerChunkSize);
+        }
+
+        @Override
+        public void close() {
+            super.close();
+            windowValues = null;
+            windowWeightValues = null;
+            influencerWeightValuesChunk.close();
+            influencerWeightValuesChunk = null;
+        }
+
+        @Override
+        public void setValueChunks(@NotNull final Chunk<? extends Values>[] valueChunks) {
+            // Cast the weight values to double
+            switch (weightChunkType) {
+                case Char:
+                    final CharChunk<?> charChunk = valueChunks[1].asCharChunk();
+                    for (int ii = 0; ii < valueChunks[1].size(); ii++) {
+                        final char val = charChunk.get(ii);
+                        influencerWeightValuesChunk.set(ii, Basic.isNull(val) ? NULL_DOUBLE : (double) val);
```

Comment:
Same opinion on the `Basic.isNull` use here. It is significantly cheaper to compare to the `QueryConstants.NULL_TYPE` since we know the type directly. Maybe we should consider adding a `Chunk#isNull(int index)` -- there are probably lots of use cases.